### PR TITLE
fix(composer): Fix to restore drag-&-drop placement after refresh

### PR DESCRIPTION
**Pull request by mumanity**
## Overview
Fix to restore drag-&-drop placement after refresh

## Legal
This project is available under the [Apache 2.0 License](http://www.apache.org/licenses/LICENSE-2.0.html).

### CODE REVIEW

**Review by jwills-jdubs — DISMISSED** [review]

File: packages/scene-composer/src/components/panels/SceneHierarchyPanel/SceneHierarchyDataProvider.tsx [hunk 1]
```diff
@@ -105,38 +104,27 @@ const sortNodes = (a, b) => {
...
+    document,
+    selectedSceneNodeRef,
+    getSceneNodeByRef,
+    setSelectedSceneNodeRef,
+    updateSceneNodeInternal,
+    getObject3DBySceneNodeRef,
+    setCameraTarget,
+    removeSceneNode,
+    isEditing,
+  } = useStore(sceneComposerId)((state) => state);
```
Comment (sheilaXu) [inline packages/scene-composer/src/components/panels/SceneHierarchyPanel/SceneHierarchyDataProvider.tsx:117]: using individual hooks is a performance improvement from https://github.com/awslabs/iot-app-kit/pull/283/files#diff-9eae39af074be35fac13ff65406838d84bce07db7e48db1e4654f15a227e23d9

File: packages/scene-composer/src/components/panels/SceneHierarchyPanel/SceneHierarchyDataProvider.tsx [hunk 2]
```diff
@@ -193,44 +190,41 @@ const SceneHierarchyDataProvider: FC<SceneHierarchyDataProviderProps> = ({ selec
...
-    [sceneComposerId],
+    [removeSceneNode],
   );
 
   return (
     <DndProvider backend={HTML5Backend}>
       <Context.Provider
         value={{
-          rootNodes: rootNodes.map((node) => toSceneHeirarchyNode(node, !isEmpty(searchTerms))),
+          rootNodes: rootNodes.map((item) => toSceneHeirarchyNode(item, item.childRefs.length > 0)),
```
Comment (sheilaXu) [inline packages/scene-composer/src/components/panels/SceneHierarchyPanel/SceneHierarchyDataProvider.tsx:227]: the hideChild flag is to hide child of the nodes found from search, so we only display a flat list of matching nodes. this change will break it

**Review by sheilaXu — CHANGES_REQUESTED** [review]

File: packages/scene-composer/src/components/panels/SceneHierarchyPanel/SceneHierarchyDataProvider.tsx (hunk 2 above)
Comment (mumanity) [reply packages/scene-composer/src/components/panels/SceneHierarchyPanel/SceneHierarchyDataProvider.tsx:221]: Local testing shows that search is working properly.

Comment (sheilaXu) [reply packages/scene-composer/src/components/panels/SceneHierarchyPanel/SceneHierarchyDataProvider.tsx:221]: you should only see the nodes matching the search, not their children (unless the children matches to)

Comment (mumanity) [reply packages/scene-composer/src/components/panels/SceneHierarchyPanel/SceneHierarchyDataProvider.tsx:221]: Current state is broken. This branch still displays children, however, it also restores the ability to search at all.

**Review by sheilaXu — APPROVED** [review]

**Pull request merged**